### PR TITLE
Add ESP32 support

### PR DESCRIPTION
**Pull request by Himura2la**
Linked PRs:
- https://github.com/cloud4rpi/docs/pull/28
- https://github.com/cloud4rpi/cloud4rpi/pull/26

### CODE REVIEW

**Review by sky3d — COMMENTED** [review]
Add steps for configure DesireTemp using Slider widget

**Review by sky3d — APPROVED** [review]

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2017 Cloud4RPi
+Copyright (c) 2017-2019 Cloud4RPi
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (dirk-pieterse) [inline LICENSE:5]: This sentence is a bit obscure. Can you explain its function?

File: README.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ The package provides a client library that simplifies connecting Internet-enable
 Currently supported hardware:
```
Comment (dirk-pieterse) [inline README.md]:
```suggestion
Supported hardware:
```

File: README.md [hunk 3]
```diff
@@ -32,51 +33,55 @@ The [Cloud4RPi library](http://platformio.org/lib/show/2045/cloud4rpi-esp-arduin
...
-10. Add a new **Chart** widget and bind it to the `Uptime` variable.
-10. Add a new **Switch** widget and bind it to the `LED On` variable.
-11. Add a new **Text** widget and bind it to the `STATUS` variable. Configure different colors for the **"IDLE"**, **"RING"** and **"BOOM!"** strings.
+3. Save the `platformio.ini` file and open the `src\main.cpp` file.
+4. Copy-paste the [sample code](examples/control/control.ino) into it. <!-- common part begins here -->
+5. [Log in to your Cloud4RPi account](https://cloud4rpi.io/signin) or [create a new one](https://cloud4rpi.io/register).
+6. Copy [your device](https://cloud4rpi.io/devices)'s **Device Token**. You can create a device on the [Devices](https://cloud4rpi.io/devices) page and copy its **Device Token** if you do not have one.
+7. Replace the `__YOUR_DEVICE_TOKEN__` string with your device token.
+8. Replace the `__SSID__` and `__PASSWORD__` strings with your Wi-Fi network data.
+9. If the `LED_BUILTIN` constant is not defined in your board's library, change this constant to the pin number to which an LED is connected (for example, `2`).
```
Comment (dirk-pieterse) [inline README.md]:
```suggestion
9. If the `LED_BUILTIN` constant is not defined in your board's library, change this constant to the LED's pin number (for example, `2`).
```

File: README.md [hunk 4]
```diff
@@ -32,51 +33,55 @@ The [Cloud4RPi library](http://platformio.org/lib/show/2045/cloud4rpi-esp-arduin
...
-6. Hit the `Verify` button and wait until the compilation is done.
-8. Connect the board in the flashing mode and hit the `Upload` button.
-9. Once flashing is complete, open `Serial Monitor` and watch the sample code's output.
-8. Notice that the [device](https://cloud4rpi.io/devices) went online and started sending data.
-9. Go to the [Control Panels](https://cloud4rpi.io/control-panels/) page and add a new control panel.
-10. Add a new **Chart** widget and bind it to the `Uptime` variable.
-10. Add a new **Switch** widget and bind it to the `LED On` variable.
-11. Add a new **Text** widget and bind it to the `STATUS` variable. Configure different colors for the **"IDLE"**, **"RING"** and **"BOOM!"** strings.
-
+    3. [Optional] Open the `%HOMEPATH%\Documents\Arduino\libraries\cloud4rpi-esp-arduino\src\Cloud4RPi.h` (`~/Documents/Arduino/libraries/cloud4rpi-esp-arduino/src/Cloud4RPi.h` on Mac) file and enable verbose output by adding the `#define CLOUD4RPI_DEBUG 1` line at the beginning.
```
Comment (dirk-pieterse) [inline README.md]: Same changes as the previous PR

**Review by dirk-pieterse — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 1 above)
Comment (Himura2la) [reply LICENSE:5]: That's the MIT license text, no need to correct it.

**Review by sky3d — APPROVED** [review]

**Review by const314 — APPROVED** [review]

**Pull request merged**